### DISCUSSION TITLE
If it is possible to develop a two-phase flow code like RELAP5

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#26182] [Q&A General]
Hi all,

I would like to know if it is possible to develop a two-phase flow code using the THM module. Just like RELAP5:
- using six-equation model;
- using the closure model in RELAP5;
- using the RDG method and Roe-type numerical method to solve convection terms;
- using water97 fluid properties;
- using the component framework in THM, and developing the two-phase components.

If it is possible? I'm not sure.....  Could anyone kindly tell me? Or I must develop our own code using the seven-equation model like RELAP7?

**Comment by GiudGiud** [comment]
Definitely possible but it's a lot of work
@joshuahansel for a more elaborate answer

**Comment by joshuahansel** [comment]
It's definitely possible, so long as you have the funding and resources to develop a two-phase code. As you suggested, RELAP-7 does the same thing but with the 7-equation model. Architecturally, you'd just follow the structure of THM or RELAP-7. If you have specific questions about how we've done things, we can answer those.

For Water97, you may need to implement more interfaces depending on your formulation and what lookups you need.

**Reply by GiudGiud** [reply]
@SomnusYu can you please give us some background on your project? Is this a classic 3 year grant for PhD research?

**Comment by SomnusYu** [comment]
Yes, it's my PHD project, and I have developed it for two years. These days I meet a lot of problems, so I am not sure it is possible to finish it.
---- Replied Message ----
| From | Guillaume ***@***.***> |
| Date | 11/29/2023 22:11 |
| To | idaholab/moose ***@***.***> |
| Cc | SomnusYu ***@***.***>,
Mention ***@***.***> |
| Subject | Re: [idaholab/moose] If it is possible to develop a two-phase flow code like RELAP5 (Discussion #26182) |

@SomnusYu can you please give us some background on your project? Is this a classic 3 year grant for PhD research?

**Reply by GiudGiud** [reply]
As for most coding endeavors I would recommend starting with simple cases. Two-phase is ambitious.

If you have a working one-phase solver and get the right answers for the relevant validation (or verification against 1D simple studies or THM) cases, you are on the right track!

As @joshuahansel mentioned we are always happy to support as much as our time allows

**Comment by SomnusYu** [comment]
Thank you very much. 
---- Replied Message ----
| From | ***@***.***> |
| Date | 11/29/2023 21:36 |
| To | idaholab/moose ***@***.***> |
| Cc | SomnusYu ***@***.***>,
Author ***@***.***> |
| Subject | Re: [idaholab/moose] If it is possible to develop a two-phase flow code like RELAP5 (Discussion #26182) |

It's definitely possible, so long as you have the funding and resources to develop a two-phase code. As you suggested, RELAP-7 does the same thing but with the 7-equation model. Architecturally, you'd just follow the structure of THM or RELAP-7. If you have specific questions about how we've done things, we can answer those.

For Water97, you may need to implement more interfaces depending on your formulation and what lookups you need.

**Comment by SomnusYu** [comment]
Thanks for you suggestions. Thank you very much!
---- Replied Message ----
| From | Guillaume ***@***.***> |
| Date | 11/29/2023 22:31 |
| To | idaholab/moose ***@***.***> |
| Cc | SomnusYu ***@***.***>,
Mention ***@***.***> |
| Subject | Re: [idaholab/moose] If it is possible to develop a two-phase flow code like RELAP5 (Discussion #26182) |

As for most coding endeavors I would recommend starting with simple cases. Two-phase is ambitious.

If you have a working one-phase solver and get the right answers for the relevant validation (or verification against 1D simple studies or THM) cases, you are on the right track!

As @joshuahansel mentioned we are always happy to support as much as our time allows
